examples/stream-collector/src/rando.cc

```diff
@@ -0,0 +1,250 @@
+                {"float32", "", ""},
+            },
+            "",
+            "float32 random number"
+        },
+        {
+            {
+                {"intel", "", ""},
+                {"cpp", "", ""},
+                {"mock", "", ""},
+                {"randomnumber", "", ""},
+                {"float64", "", ""},
+            },
+            "",
+            "float64 random number"
+        },
+        {
+            {
+                {"intel", "", ""},
+                {"cpp", "", ""},
+                {"mock", "", ""},
+                {"randomnumber", "", ""},
+                {"int32", "", ""},
+            },
+            "",
+            "int32 random number"
+            },
+        {
+            {
+                {"intel", "", ""},
+                {"cpp", "", ""},
+                {"mock", "", ""},
+                {"randomnumber", "", ""},
+                {"int64", "", ""},
+            },
+            "",
+            "int64 random number"
+        },
+        {
+            {
+                {"intel", "", ""},
+                {"cpp", "", ""},
+                {"mock", "", ""},
+                {"randomnumber", "", ""},
+                {"uint32", "", ""},
+            },
+            "",
+            "uint32 random number"
+        },
+        {
+            {
+                {"intel", "", ""},
+                {"cpp", "", ""},
+                {"mock", "", ""},
+                {"randomnumber", "", ""},
+                {"uint64", "", ""},
+            },
+            "",
+            "uint64 random number"
+        },
+        {
+            {
+                {"intel", "", ""},
+                {"cpp", "", ""},
+                {"mock", "", ""},
+                {"randomboolean", "", ""},
+                {"boolean", "", ""},
+            },
+            "",
+            "random boolean"
+        },
+        {
+            {
+                {"intel", "", ""},
+                {"cpp", "", ""},
+                {"mock", "", ""},
+                {"randomstring", "", ""},
+                {"string", "", ""},
+            },
+            "",
+            "random string"
+        }
+    };
+    return metrics;
+}
+
+void Rando::stream_metrics(std::vector<Plugin::Metric> &metsIn,
+                            std::vector<Plugin::Metric> &metsOut,
+                            std::string &errMsg) {
+    auto stream = std::async(std::launch::async, &Rando::stream_it,
+                            this, std::ref(metsOut), std::ref(errMsg));
+}
+
+void Rando::stream_it(std::vector<Plugin::Metric> &metsOut, std::string &errMsg) {
+    while (1) {
+        if (!metsOut.empty()) {
+            std::vector<Metric>::iterator mets_iter;
+            unsigned int seed = time(NULL);
+            int random_value = rand_r(&seed) % 1000;
+
+            for (mets_iter = metsOut.begin(); mets_iter != metsOut.end(); mets_iter++) {
+                std::string ns_mts_type = mets_iter->ns()[4].value;
```

Comment:
You can use recently introduced `get_value()` method  to get value of namespace element.